core/arch/arm/kernel/user_ta.c

```diff
@@ -437,6 +438,68 @@ static void user_ta_dump_state(struct tee_ta_ctx *ctx)
 }
 KEEP_PAGER(user_ta_dump_state);
 
+#ifdef CFG_TA_FTRACE_SUPPORT
+static void user_ta_dump_state_buffer(struct tee_ta_ctx *ctx, char *buf,
```

Comment:
Maybe clarify with a comment (here or next to the function prototype in header file) what is expected from in/out argument `sz`. Looks like input value is the buffer size and output value is the buffer remaining size.
A more standard way would be to output the effectively written buffer size rather than the remaining buffer size.

Reply:
Its effectively written buffer size only rather than the remaining buffer size.